pkg/controller/job/job_controller.go

```diff
@@ -1871,8 +1871,10 @@ func countReadyPods(pods []*v1.Pod) int32 {
 // PodReplacementPolicy controls when we recreate pods if they are marked as terminating
 // Failed means that we recreate only once the pod has terminated.
 func onlyReplaceFailedPods(job *batch.Job) bool {
-	if feature.DefaultFeatureGate.Enabled(features.JobPodReplacementPolicy) && *job.Spec.PodReplacementPolicy == batch.Failed {
-		return true
+	if feature.DefaultFeatureGate.Enabled(features.JobPodReplacementPolicy) {
```

Comment:
TBH I don't really understand why this could happen but I was getting a segmentation fault on the negative test case for the test I added.  

Checking feature first and then checking the podreplacement policy solved this.

Reply:
I researched this and I opened up PRs to fix this.

Reply:
PRs were merged so this is irrelevant.